ext/php8/ip_extraction.c

```diff
@@ -446,7 +446,10 @@ static bool dd_parse_ip_address_maybe_port_pair(const char *addr, size_t addr_le
     }
     const char *colon = memchr(addr, ':', addr_len);
     if (colon) {
-        return dd_parse_ip_address(addr, colon - addr, out);
+        size_t colon_position = colon - addr;
+        if (colon_position + 1 < addr_len && !memchr(&colon[1], ':', &addr[addr_len] - &colon[1])) {
+            return dd_parse_ip_address(addr, colon_position, out);
+        }
```

Comment:
Alternatively you can call `memchr` and `memrchr` and if they don't coincide then you have two colons.

Reply:
Oh, that's one is a nice one 👏  I'll do it now

Reply:
Done